### PR TITLE
fix bio-metric login

### LINKED ISSUE #19277: Can't login with biometric ID
# Bug Report

## Problem 

Screenshot 2024-03-15 at 13 06 01

### Reproduction

1. Create a user, enable biometric ID on onboarding
2. Close the app
3. Open the app, try to log in

### Additional Information

- Status version: nightly 15/05
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/19277

status: ready

### CODE REVIEW

File: src/status_im/contexts/profile/login/events.cljs [hunk 1]
```diff
@@ -177,7 +177,7 @@
  :profile.login/get-user-password-success
  (fn [{:keys [db]} [password]]
    (when password
-     (let [{:keys [key-uid password]} (:profile/login db)]
+     (let [{:keys [key-uid]} (:profile/login db)]
```
Comment (flexsurfer) [inline src/status_im/contexts/profile/login/events.cljs]: that looks strange, do you know why?

Comment (Parveshdhull) [reply src/status_im/contexts/profile/login/events.cljs]: Hi @flexsurfer, Thank you for reviewing PR.

Removed let block, is it looks ok now? (Or you meant something else)

Comment (Parveshdhull) [reply src/status_im/contexts/profile/login/events.cljs]: btw, interestingly we had somewhat same code before [refactor](https://github.com/status-im/status-mobile/pull/19005), but due to rf/merge password was updated before using

main
main2

File: src/status_im/contexts/profile/login/events.cljs [hunk 2]
```diff
@@ -177,12 +177,13 @@
  :profile.login/get-user-password-success
  (fn [{:keys [db]} [password]]
    (when password
-     (let [{:keys [key-uid password]} (:profile/login db)]
-       {:db (-> db
-                (assoc-in [:profile/login :password] password)
-                (assoc-in [:profile/login :processing] true))
-        :fx [[:set-root :progress]
-             [:effects.profile/login [key-uid (security/safe-unmask-data password)]]]}))))
+     {:db (-> db
```
Comment (flexsurfer) [inline src/status_im/contexts/profile/login/events.cljs:180]: but now there is no change? how is it fixed then?

Comment (Parveshdhull) [reply src/status_im/contexts/profile/login/events.cljs:180]: event `get-user-password-success` was called with right password. But in let block we were overwriting it with nil value

**Review by flexsurfer — APPROVED** [review]

**Review by alwx — APPROVED** [review]

**Comment by mariia-skrypnyk** [comment]
Hi @Parveshdhull !

Thanks for your PR. 
Checked such scenarios on Android/iOS:

✅Onboarding enabling and login
✅Onboarding disabling and switch on bio-metric from Profile and login
✅Wallet Account creation
✅New user bio-metric login
✅Recovered user bio-metric login
✅Logout and no bio-metric login

Thanks for your fix. PR can be merged.

**Pull request merged**